src/libCli/Completion.cpp

```diff
@@ -19,198 +19,251 @@ using namespace ArgParse;
+    {
+        // completion requested :)
+        if (f_debug)
         {
-          start = 0;
+            std::cerr << "Input string \"" << f_args << std::endl;
         }
-        end = candidateStr.find_first_of(" ", n)-1;
 
-        if(f_debug)
+        //size_t n = parseTree.getMatchedString().size();
+        size_t n = f_args.size();
+        for (auto candidate : f_candidates)
         {
-          printf("nospace! cand='%s', n=%zu, start=%zu, end = %zu\n",candidateStr.c_str(), n, start, end);
-        }
+            std::string candidateStr = candidate->getMatchedStringRaw();
 
-        suggestion = candidateStr.substr(start, std::string::npos);
-        size_t trimEnd = suggestion.find_last_not_of(' ');
-        suggestion = suggestion.substr(0, trimEnd+1);
-        //suggestion = candidateStr.substr(start, end-start+1);
+            if (f_debug)
+            {
+                std::cout << "------------------------------------------------" << std::endl;
+                std::cout << "candiate string: " << candidateStr << std::endl;
+                std::cout << "candiate's debug string: " << std::endl;
+                std::cout << candidate->getDebugString() << std::endl;
+                std::cout << "*************Suggestion****************";
+                std::cout << "------------------------------------------------" << std::endl;
+            }
 
-        if(!suggestionDoc.empty())
-        {
-            suggestion = suggestion + "\t" + suggestionDoc;
-        }
+            std::string suggestion;
+            size_t start = n;
+            size_t end;
 
-        if(f_debug)
-        {
-            printf("post: '%s'\n", suggestion.c_str());
+            std::string suggestionDoc = candidate.get()->getShortDocument();
+            std::string suggestionDocRoot = f_parseTree.getShortDocument();
+            /// here the document (description of the tab completion) extracted from the original parsed tree and from the candidate tree should be compared,
+            /// if there is no new document of candidate in the parsed tree, that means the both extracted documents must same,
+            ///we only find the document at the right most position in the parsed tree (the newest) for tab completion.
+            if (suggestionDocRoot == suggestionDoc)
+            {
+                suggestionDoc = "";
+            }
+
+            if (f_debug)
+            {
+                std::cout << "suggestionDoc: " << suggestionDoc << std::endl;
+                printf("pre: '%s'\n", candidateStr.c_str());
+                printf("candidateStr[n=%zu] = '%c'\n", n, candidateStr[n]);
+            }
+
+            //bis char 5 completen --> Leer wird completed
+            std::vector<std::string> suggestions;
+            suggestion = getLastArgument(candidateStr, " ", suggestions);
```

Comment:
as discussed:
we need some way of distinguishing old (which user already typed) and new (completions) content.
We then need to either return
- Current "token" starting from last space the user entered ending at end of this token (with or without new information) if last char from user was not a space
- next token if user already entered a space

Also take note, that a cadidate might contain multiple "new" tokens